pkg/provider/kubernetes/client.go

```diff
@@ -69,6 +108,18 @@ func (c *Client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretD
 	if err != nil {
 		return nil, err
 	}
+	if ref.MetadataPolicy == esv1beta1.ExternalSecretMetadataPolicyFetch {
+		retData := make(map[string][]byte)
```

Comment:
`GetSecret()` above supports `ref.Property` which is needed to target either labels or annotations.
I think it would be a huge plus if this would be supported here as well:

i.e. if `ref.Property` is `labels` then all labels are returns as map (same goes for annotations ofc).

Reply:
I guess we don't need to support gjson here b/c labels and annotations are only key/value pairs.

Reply:
Hey Moritz, let me rephrase this to see if I understood correctly. What you're saying is that if I have a DataFrom as follows
```yaml
   dataFrom:
   - extract:
       key: json-secret
       property: label
```
You would end up with a secret with every label (and the same for annotations). Is that correct?

Reply:
How do we know we're talking about labels and not values? do we want to support `metadataPolicy: Fetch` also in dataFrom?

Reply:
:point_up_2: right, what you've posted plus the `metadataPolicy=Fetch`. I'd also make the `property` value plural: `labels` and `annotations`.
```yaml
   dataFrom:
   - extract:
       key: k8s-source-secret
       metadataPolicy: Fetch
       property: labels
```

Reply:
I tested with this

```yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: k8s-example
spec:
  refreshInterval: 1h
  secretStoreRef:
    kind: SecretStore
    name: k8s-store
  target:
    name: k8s-example
  dataFrom:
  - extract:
      key: k8s-source-secret
      metadataPolicy: Fetch
      property: labels
```

And expected a secret that contains the label as key/value pairs:

say a secret has 

```yaml
metadata:
  labels:
    example: "baz"
    foo: "bar"
```

I would want to have this:
```yaml
data:
  example: "..."
  foo: "..."
```

Instead i get a json-encoded value for the key `labels`.

```yaml
apiVersion: v1
kind: Secret
data:
  # resembles to: {"example":"baz","foo":"bar"}
  labels: eyJleGFtcGxlIjoiYmF6IiwiZm9vIjoiYmFyIn0=
```

Reply:
I thought in this case that was expected... so it's exactly the same from when using it in the `data:`
No problem, I'll fix it